### PR TITLE
Miscellaneous improvements.

### PR DESCRIPTION
**Pull request by drug007**

### CODE REVIEW

File: source/asdf/serialization.d [hunk 1]
```diff
@@ -1981,7 +2052,8 @@ void deserializeValue(V : T[], T)(Asdf data, ref V value)
 			value = null;
 			return;
 		default:
-			throw new DeserializationException(kind);
+			import std.conv : text;
```
Comment (9il) [inline source/asdf/serialization.d]: please remove this commit. I would like to minimize string allocation with future @nogc exceptions in mind

File: source/asdf/serialization.d [hunk 2]
```diff
@@ -1465,6 +1465,13 @@ void serializeValue(S)(ref S serializer, bool value)
 	serializer.putValue(value);
 }
 
+/// Char serialization
+void serializeValue(S, V)(ref S serializer, V value) if (isSomeChar!V)
```
Comment (9il) [inline source/asdf/serialization.d]: Should be just `void serializeValue(S)(ref S serializer, char value)`

Comment (drug007) [reply source/asdf/serialization.d]: I've done it initially but then I think about wchar and dchar. Wouldn't it be useful too?

Comment (9il) [reply source/asdf/serialization.d]: They can not be converted to char[1]

Comment (drug007) [reply source/asdf/serialization.d]: I use `V[1]` now.

Comment (9il) [reply source/asdf/serialization.d]: asdf does not support wstring, dstring

Comment (drug007) [reply source/asdf/serialization.d]: I see. Fixed.

File: source/asdf/serialization.d [hunk 3]
```diff
@@ -1927,6 +1990,13 @@ void deserializeValue(V)(Asdf data, ref V value)
 	}
 }
 
+/// Deserialize single char
+void deserializeValue(V)(Asdf data, V value) if (isSomeChar!V)
```
[the remark this answers is not in the document]
Comment (drug007) [reply source/asdf/serialization.d]: Ok. I'm going to return to this PR in the next days to complete it.

**Pull request merged**